packages/web-components/src/components/ic-text-field/ic-text-field.tsx

```diff
@@ -331,16 +339,19 @@ export class TextField {
   componentWillLoad(): void {
     if (this.value !== this.initialValue) {
       this.watchValueHandler(this.value);
+    } else {
+      if (this.maxCharacters > 0) {
+        this.value = this.value.substring(0, this.maxCharacters);
+      }
     }
 
-    this.getMaxLengthExceeded(this.value);
+    this.getMaxValueExceeded(this.value);
```

Comment:
just realized this is slightly inefficient as both `this.getMaxValueExceeded` and `this.getMaxCharactersReached` may have already been called due to the `watchValueHandler` being called directly (line 341), or indirectly when `this.value` is changed on line 344

Reply:
I can't seem to get it to work...

I tried moving those two methods into an `else` on line 344, but then [tests were failing](https://github.com/mi6/ic-ui-kit/actions/runs/9940487837/job/27457312100). It was like line 343 was not causing the `watchValueHandler` to be called at all to set the character count correctly.

So I'm a bit confused 😅 don't know if it's just best to keep it as is even though it is a bit inefficient